library/src/main/java/com/datatorrent/lib/io/jms/JMSBase.java

```diff
@@ -355,10 +366,13 @@ public void createConnection() throws JMSException
    */
   protected ConnectionFactory getConnectionFactory()
   {
-    logger.debug("class {} properties {}", connectionFactoryClass, connectionFactoryProperties);
+    logger.debug("connectionFactoryBuilder {} connectionFactoryClass {} properties {}", "" + connectionFactoryBuilder,
+        connectionFactoryClass, connectionFactoryProperties);
     ConnectionFactory cf;
     try {
-      if (connectionFactoryClass != null) {
+      if (connectionFactoryBuilder != null) {
+        cf = connectionFactoryBuilder.buildConnectionFactory();
+      } else if (connectionFactoryClass != null) {
```

Comment:
Also shouldn't the connection factory properties be actually handled by the builder?